### ISSUE TITLE
The fair random generator

### ISSUE THREAD

**Issue by siers** [vicgeralds/vitetris#7]
I've wondered for a while whether it would be possible to make the tetromino random generator  a bit more fair like described in [tetris.fandom.com/…/Random_generator](https://tetris.fandom.com/wiki/Random_Generator)?

Would you be interested in implementing this feature?

**Comment by vicgeralds** [comment]
I actually think I implemented the 7 bag random generator some time quite long ago. I'll see if I can find that on some old drive and what could be done with it.

In general though I'm not so much interested in adding new features (that might infringe on copyrights) to vitetris myself.

**Comment by siers** [comment]
Is it actually copyrighted?

**Comment by vicgeralds** [comment]
I don't think trying to answer that question is going to help. My point was that I don't really have any plans to develop more features and that I'm a bit cautious when it comes to copying game elements.

But it turns out that I already implemented a random generator like this and used it in a time attack mode back in 2009.

**Comment by vicgeralds** [comment]
Check this out: d0473a54d57aaf068d99f591d13f264d9a46bde7
I don't want all these changes to end up in master, so I'm not quite sure what to do with it.

Adding just the random generator as an option somehow seems a bit odd to me as it would change the difficulty (making it easier to get higher scores). What do you think?

**Comment by siers** [comment]
Yeah, that has quite a few changes together with the new random generator code. Perhaps you could remove the random generator bit, if that makes you hesitant to merge it, and then add it to master.

But hey, at least now I can check this version out for myself to play with the bag-based RNG. :P 

P.S. Turns out there's even another, supposedly more advanced elaboration of the tetris RNG: https://simon.lc/the-history-of-tetris-randomizers.

**Comment by vicgeralds** [comment]
That's some interesting reading, thanks!
